diff --git a/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go b/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go
index da34d64f..a8c3fc1f 100644
--- a/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go
+++ b/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go
@@ -22,9 +22,10 @@ func newRemoteAuthenticator(tokenReview authenticationclient.TokenReviewInterfac
 	authenticators := []authenticator.Request{}
 
 	// TODO audiences
-	// Upstream k8s.io introduced a configurable timeout for the token request - it previously defaulted to 10 seconds.
-	// See https://github.com/kubernetes/kubernetes/pull/100959
-	tokenAuthenticator, err := webhooktoken.NewFromInterface(tokenReview, nil, *webhooktoken.DefaultRetryBackoff(), 10*time.Second)
+	// Upstream kubernetes/apiserver introduced timeouts in https://github.com/kubernetes/kubernetes/pull/100959.
+	// This change is included in v0.20.9 (used by openshift-controller-manager) but not v0.20.4 (used by openshift-apiserver).
+	// Breaking the tie by reverting 4ff13dee21a76d7bd107426f18b0e560aa2c94ed.
+	tokenAuthenticator, err := webhooktoken.NewFromInterface(tokenReview, nil, *webhooktoken.DefaultRetryBackoff())
 	if err != nil {
 		return nil, err
 	}
