diff --git a/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go b/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go
index ea69b8b8..8b5049b1 100644
--- a/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go
+++ b/vendor/github.com/openshift/openshift-controller-manager/pkg/cmd/controller/apiserver_authenticator.go
@@ -22,7 +22,8 @@ func newRemoteAuthenticator(tokenReview authenticationclient.TokenReviewInterfac
 	authenticators := []authenticator.Request{}
 
 	// TODO audiences
-	tokenAuthenticator, err := webhooktoken.NewFromInterface(tokenReview, nil, *webhooktoken.DefaultRetryBackoff())
+	TokenAccessReviewTimeout := 10 * time.Second
+	tokenAuthenticator, err := webhooktoken.NewFromInterface(tokenReview, nil, *webhooktoken.DefaultRetryBackoff(), TokenAccessReviewTimeout)
 	if err != nil {
 		return nil, err
 	}
