diff --git a/vendor/github.com/openshift/oauth-apiserver/pkg/version/version.go b/vendor/github.com/openshift/oauth-apiserver/pkg/version/version.go
index 62a41565..1b021088 100644
--- a/vendor/github.com/openshift/oauth-apiserver/pkg/version/version.go
+++ b/vendor/github.com/openshift/oauth-apiserver/pkg/version/version.go
@@ -46,7 +46,7 @@ func Get() version.Info {
 func init() {
 	buildInfo := metrics.NewGaugeVec(
 		&metrics.GaugeOpts{
-			Name: "openshift_build_info",
+			Name: "openshift_build_info_oauth",
 			Help: "A metric with a constant '1' value labeled by major, minor, git commit & git version from which OpenShift was built.",
 		},
 		[]string{"major", "minor", "gitCommit", "gitVersion"},
