commit 10c701fdc61a60d1efef018b54ad2470c0b52d10
Author: rootfs <hchen@redhat.com>
Date:   Sat May 8 13:28:15 2021 -0400

    hack: disable stopCh - it will be double closed

diff --git a/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go b/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go
index 9a3d04c7..8dc72f01 100644
--- a/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go
+++ b/vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go
@@ -126,8 +126,8 @@ controller, and serviceaccounts controller.`,
 				os.Exit(1)
 			}
 
-			stopCh := server.SetupSignalHandler(true)
-			if err := Run(c.Complete(), stopCh); err != nil {
+			//stopCh := server.SetupSignalHandler(true)
+			if err := Run(c.Complete(), nil /*stopCh*/); err != nil {
 				fmt.Fprintf(os.Stderr, "%v\n", err)
 				os.Exit(1)
 			}
