commit ccdbe2b7efff802c55b2131078faa19c5521dd3b
Author: rootfs <hchen@redhat.com>
Date:   Thu May 13 13:19:49 2021 -0400

    hack: avoid double close

diff --git a/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go b/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go
index 48d31a67..0573b9be 100644
--- a/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go
+++ b/vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go
@@ -123,8 +123,8 @@ func runCommand(cmd *cobra.Command, opts *options.Options, registryOptions ...Op
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	go func() {
-		//stopCh := server.SetupSignalHandler(true)
-		//<-stopCh
+		stopCh := make(chan struct {}) //server.SetupSignalHandler(true)
+		<-stopCh
 		cancel()
 	}()
 
