commit 443635dd15fd016fa1a3100372246f46f08f0e2f
Author: rootfs <hchen@redhat.com>
Date:   Tue Jun 15 10:54:05 2021 -0400

    hack: allow signal handling to terminate program

diff --git a/vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go b/vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go
index 4cb510b..b51892a 100644
--- a/vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go
+++ b/vendor/k8s.io/kubernetes/cmd/kube-apiserver/app/server.go
@@ -178,7 +178,7 @@ cluster's shared state through which all other components interact.`,
 				return utilerrors.NewAggregate(errs)
 			}
 
-			return Run(completedOptions, genericapiserver.SetupSignalHandler(false))
+			return Run(completedOptions, genericapiserver.SetupSignalHandler(true))
 		},
 		Args: func(cmd *cobra.Command, args []string) error {
 			for _, arg := range args {
