diff --git a/deps/k8s.io/klog/goroutine_labels.go b/deps/k8s.io/klog/goroutine_labels.go
new file mode 100644
index 00000000..50f2235b
--- /dev/null
+++ b/deps/k8s.io/klog/goroutine_labels.go
@@ -0,0 +1,7 @@
+package klog
+
+import "k8s.io/klog/v2/internal/buffer"
+
+func WithMicroshiftLoggerComponent(c string, f func()) {
+	buffer.WithMicroshiftLoggerComponent(c, f)
+}
diff --git a/deps/k8s.io/klog/internal/buffer/buffer.go b/deps/k8s.io/klog/internal/buffer/buffer.go
index ac88682a..a29eb626 100644
--- a/deps/k8s.io/klog/internal/buffer/buffer.go
+++ b/deps/k8s.io/klog/internal/buffer/buffer.go
@@ -149,6 +149,8 @@ func (buf *Buffer) FormatHeader(s severity.Severity, file string, line int, now
 	buf.Tmp[21] = ' '
 	buf.nDigits(7, 22, Pid, ' ') // TODO: should be TID
 	buf.Tmp[29] = ' '
+	buf.WriteString(getMicroshiftLoggerComponent())
+	buf.WriteByte(' ')
 	buf.Write(buf.Tmp[:30])
 	buf.WriteString(file)
 	buf.Tmp[0] = ':'
diff --git a/deps/k8s.io/klog/internal/buffer/goroutine_labels.go b/deps/k8s.io/klog/internal/buffer/goroutine_labels.go
new file mode 100644
index 00000000..6e93bdbd
--- /dev/null
+++ b/deps/k8s.io/klog/internal/buffer/goroutine_labels.go
@@ -0,0 +1,30 @@
+package buffer
+
+import (
+	"context"
+	"runtime/pprof"
+	"unsafe"
+)
+
+//go:linkname runtime_getProfLabel runtime/pprof.runtime_getProfLabel
+func runtime_getProfLabel() unsafe.Pointer
+
+func getMicroshiftLoggerComponent() string {
+	labels := (*map[string]string)(runtime_getProfLabel())
+	if labels == nil {
+		return "???"
+	}
+
+	c, ok := (*labels)["microshift_logger_component"]
+	if !ok {
+		return "???"
+	}
+
+	return c
+}
+
+func WithMicroshiftLoggerComponent(c string, f func()) {
+	pprof.Do(context.Background(), pprof.Labels("microshift_logger_component", c), func(context.Context) {
+		f()
+	})
+}
