diff --git a/deps/github.com/openshift/kubernetes/cmd/kube-controller-manager/app/controllermanager.go b/deps/github.com/openshift/kubernetes/cmd/kube-controller-manager/app/controllermanager.go
index 76f30faf8..3b3682753 100644
--- a/deps/github.com/openshift/kubernetes/cmd/kube-controller-manager/app/controllermanager.go
+++ b/deps/github.com/openshift/kubernetes/cmd/kube-controller-manager/app/controllermanager.go
@@ -38,7 +38,6 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/uuid"
 	"k8s.io/apimachinery/pkg/util/wait"
-	"k8s.io/apiserver/pkg/server"
 	"k8s.io/apiserver/pkg/server/healthz"
 	"k8s.io/apiserver/pkg/server/mux"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
@@ -159,8 +158,7 @@ controller, and serviceaccounts controller.`,
 			fg.(featuregate.MutableFeatureGate).AddMetrics()
 			// add component version metrics
 			s.ComponentGlobalsRegistry.AddMetrics()
-			stopCh := server.SetupSignalHandler()
-			return Run(context.Background(), c.Complete(), stopCh)
+			return Run(cmd.Context(), c.Complete(), cmd.Context().Done())
 		},
 		Args: func(cmd *cobra.Command, args []string) error {
 			for _, arg := range args {
