diff --git a/vendor/k8s.io/kubernetes/pkg/serviceaccount/claims.go b/vendor/k8s.io/kubernetes/pkg/serviceaccount/claims.go
index fa195b04..08f21361 100644
--- a/vendor/k8s.io/kubernetes/pkg/serviceaccount/claims.go
+++ b/vendor/k8s.io/kubernetes/pkg/serviceaccount/claims.go
@@ -90,7 +90,7 @@ func Claims(sa core.ServiceAccount, pod *core.Pod, secret *core.Secret, expirati
 	}
 
 	if warnafter != 0 {
-		pc.Kubernetes.WarnAfter = jwt.NewNumericDate(now.Add(time.Duration(warnafter) * time.Second))
+		pc.Kubernetes.WarnAfter = *jwt.NewNumericDate(now.Add(time.Duration(warnafter) * time.Second))
 	}
 
 	return sc, pc
