diff --git a/deps/github.com/openshift/route-controller-manager/pkg/cmd/route-controller-manager/cmd.go b/deps/github.com/openshift/route-controller-manager/pkg/cmd/route-controller-manager/cmd.go
index afb7dc267..e255d16bd 100644
--- a/deps/github.com/openshift/route-controller-manager/pkg/cmd/route-controller-manager/cmd.go
+++ b/deps/github.com/openshift/route-controller-manager/pkg/cmd/route-controller-manager/cmd.go
@@ -6,6 +6,7 @@ import (
 	"os"
 
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/clock"
 
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
 
@@ -18,7 +19,7 @@ const (
 )
 
 func NewRouteControllerManagerCommand(name string) *cobra.Command {
-	cmd := controllercmd.NewControllerCommandConfig("route-controller-manager", rcmversion.Get(), RunRouteControllerManager).
+	cmd := controllercmd.NewControllerCommandConfig("route-controller-manager", rcmversion.Get(), RunRouteControllerManager, clock.RealClock{}).
 		WithComponentOwnerReference(&corev1.ObjectReference{
 			Kind:      "Pod",
 			Name:      os.Getenv(podNameEnv),
