diff --git a/deps/github.com/openshift/cluster-policy-controller/pkg/cmd/cluster-policy-controller/openshiftcontrolplane_config.go b/deps/github.com/openshift/cluster-policy-controller/pkg/cmd/cluster-policy-controller/openshiftcontrolplane_config.go
index e8dcea272..1a4a2f944 100644
--- a/deps/github.com/openshift/cluster-policy-controller/pkg/cmd/cluster-policy-controller/openshiftcontrolplane_config.go
+++ b/deps/github.com/openshift/cluster-policy-controller/pkg/cmd/cluster-policy-controller/openshiftcontrolplane_config.go
@@ -28,8 +28,6 @@ func asOpenshiftControllerManagerConfig(config *unstructured.Unstructured) (*ope
 }
 
 func setRecommendedOpenShiftControllerConfigDefaults(config *openshiftcontrolplanev1.OpenShiftControllerManagerConfig) {
-	configdefaults.SetRecommendedKubeClientConfigDefaults(&config.KubeClientConfig)
-
 	configdefaults.DefaultStringSlice(&config.Controllers, []string{"*"})
 
 	configdefaults.DefaultString(&config.SecurityAllocator.UIDAllocatorRange, "1000000000-1999999999/10000")
