diff --git a/etcd/cmd/microshift-etcd/run.go b/etcd/cmd/microshift-etcd/run.go
index 018b129a7..a573a5c01 100644
--- a/etcd/cmd/microshift-etcd/run.go
+++ b/etcd/cmd/microshift-etcd/run.go
@@ -19,7 +19,7 @@ import (
 
 	"github.com/spf13/cobra"
 	etcd "go.etcd.io/etcd/server/v3/embed"
-	"go.etcd.io/etcd/server/v3/mvcc/backend"
+	"go.etcd.io/etcd/server/v3/storage/backend"
 	"k8s.io/klog/v2"
 )
 
@@ -100,7 +100,7 @@ func (s *EtcdService) configure(cfg *config.Config) {
 	s.etcdCfg.PeerTLSInfo.KeyFile = cryptomaterial.PeerKeyPath(etcdPeerCertDir)
 	s.etcdCfg.PeerTLSInfo.TrustedCAFile = etcdSignerCertPath
 
-	s.etcdCfg.ExperimentalMaxLearners = MaxLearners
+	s.etcdCfg.MaxLearners = MaxLearners
 
 	updateConfigFromFile(s.etcdCfg, getConfigFilePath())
 }
diff --git a/etcd/go.mod b/etcd/go.mod
index b1c7f1fea..d1d486f02 100644
--- a/etcd/go.mod
+++ b/etcd/go.mod
@@ -154,7 +154,6 @@ replace (
 	go.etcd.io/etcd/client/pkg/v3 => github.com/openshift/etcd/client/pkg/v3 v3.5.1-0.20251027135311-239c469ebe0b // from etcd
 	go.etcd.io/etcd/client/v3 => github.com/openshift/etcd/client/v3 v3.5.1-0.20251027135311-239c469ebe0b // from etcd
 	go.etcd.io/etcd/pkg/v3 => github.com/openshift/etcd/pkg/v3 v3.5.1-0.20251027135311-239c469ebe0b // from etcd
-	go.etcd.io/etcd/raft/v3 => github.com/openshift/etcd/raft/v3 v3.5.1-0.20251027135311-239c469ebe0b // from etcd
 	go.etcd.io/etcd/server/v3 => github.com/openshift/etcd/server/v3 v3.5.1-0.20251027135311-239c469ebe0b // from etcd
 )
 
